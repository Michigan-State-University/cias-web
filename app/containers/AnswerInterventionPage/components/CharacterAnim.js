import React, { useEffect, useReducer } from 'react';
import PropTypes from 'prop-types';
import isNumber from 'lodash/isNumber';
import get from 'lodash/get';
import Lottie from 'react-lottie';

import useDidUpdateEffect from 'utils/useDidUpdateEffect';
import useMoveHelper from 'utils/animationsHelpers/useMoveHelper';
import useAudioHelper from 'utils/animationsHelpers/useAudioHelper';
import useAnimationHelper from 'utils/animationsHelpers/useAnimationHelper';
import { useAsync } from 'utils/useAsync';
import {
  bodyAnimationType,
  speechType,
  headAnimationType,
  reflectionType,
  readQuestionBlockType,
  pauseType,
  feedbackBlockType,
  reflectionFormulaType,
} from 'models/Narrator/BlockTypes';

import Draggable from 'react-draggable';
import usePauseHelper from 'utils/animationsHelpers/usePauseHelper';
import { NarratorContainer } from './styled';
import useFeedbackHelper from '../animationsHelpers/feedbackHelper';

const UPDATE = 'UPDATE';

const reducer = (state, action) => {
  const { type, newState } = action;
  switch (type) {
    case UPDATE:
      return { ...state, ...newState };
    default:
      return state;
  }
};

const initialState = {
  currentData: null,
  currentBlockIndex: 0,
};

const lottieStyles = {
  margin: 'none',
};

const CharacterAnim = ({
  blocks,
  questionId,
  settings,
  animationContainer,
  previewMode,
  answers,
  changeIsAnimationOngoing,
  setFeedbackSettings,
  feedbackScreenSettings: { sliderRef },
  audioInstance,
}) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const dispatchUpdate = newState =>
    dispatch({
      type: UPDATE,
      newState,
    });

  const changeBlock = async prevIndex => {
    clearAnimationBlock();

    cleanAudio();
    const nextIndex =
      (isNumber(prevIndex) ? prevIndex : state.currentBlockIndex) + 1;
    const nextBlock = blocks[nextIndex];
    if (nextBlock) {
      const newPosition = calculatePosition(
        nextBlock.action,
        nextBlock.endPosition,
      );

      await moveAnimation({ ...nextBlock, endPosition: newPosition });
      switch (nextBlock.type) {
        case bodyAnimationType:
        case headAnimationType:
          changeAnimation(nextBlock, nextIndex);
          break;

        case speechType:
        case reflectionType:
        case readQuestionBlockType:
        case reflectionFormulaType:
          changeSpeech(nextBlock, nextIndex);
          break;

        case pauseType:
          changePauseBlock(nextBlock, nextIndex);
          break;

        case feedbackBlockType:
          changeFeedback(nextBlock, nextIndex);
          break;

        default:
          await changeBlock(nextIndex);
          break;
      }
    } else {
      const nextAnim = getIdleAnimation();
      dispatchUpdate({
        currentData: nextAnim,
        currentBlockIndex: nextIndex,
      });
    }
  };

  const {
    getInitialBodyOrHeadAnimation,
    changeAnimation,
    handleBodyOrHeadAnimationBlock,
    getIdleAnimation,
    clearAnimationBlock,
    animationRef,
    fetchBodyAndHeadAnimations,
    loadedAnimations,
  } = useAnimationHelper(
    blocks,
    dispatchUpdate,
    changeBlock,
    state.currentData,
  );

  const {
    handlePauseBlock,
    getInitialPauseAnimation,
    changePauseBlock,
  } = usePauseHelper(
    blocks,
    state.currentData,
    dispatchUpdate,
    changeBlock,
    getIdleAnimation,
  );

  const {
    changeSpeech,
    getInitialSpeechAnimation,
    cleanAudio,
    handleAudioBlock,
    decideIfPlaySpeechAnimation,
    fetchAudioAnimations,
    stopSpeech,
  } = useAudioHelper(
    blocks,
    dispatchUpdate,
    state.currentData,
    state.currentBlockIndex,
    animationRef.current,
    changeBlock,
    answers,
    settings,
    audioInstance,
  );

  const {
    getInitialFeedbackData,
    changeFeedback,
    handleFeedbackBlock,
    calculatePosition,
  } = useFeedbackHelper(
    blocks,
    dispatchUpdate,
    changeBlock,
    state.currentData,
    setFeedbackSettings,
    sliderRef,
    animationRef.current,
    loadedAnimations,
  );

  const { animationPos, moveAnimation, fetchMoveAnimations } = useMoveHelper(
    animationContainer,
    blocks,
    dispatchUpdate,
  );

  const getInitialData = () => {
    if (blocks.length) {
      switch (blocks[0].type) {
        case speechType:
        case reflectionType:
        case readQuestionBlockType:
        case reflectionFormulaType:
          return getInitialSpeechAnimation();

        case headAnimationType:
        case bodyAnimationType:
          return getInitialBodyOrHeadAnimation();

        case pauseType:
          return getInitialPauseAnimation();

        case feedbackBlockType:
          return getInitialFeedbackData();

        default:
          break;
      }
    }
  };
  const fetchData = async () => {
    if (blocks[0]) {
      const newPosition = calculatePosition(
        blocks[0].action,
        blocks[0].endPosition,
      );

      await moveAnimation({ ...blocks[0], endPosition: newPosition });
    }
    await fetchBodyAndHeadAnimations();
    await fetchAudioAnimations();
    await fetchMoveAnimations();
  };

  useAsync(
    fetchData,
    () =>
      dispatchUpdate({
        currentData: getInitialData(),
        currentBlockIndex: 0,
      }),
    { useDeps: [questionId, previewMode], cleanUpFunction: stopSpeech },
  );

  useDidUpdateEffect(() => {
    if (state.currentData)
      switch (state.currentData.type) {
        case headAnimationType:
        case bodyAnimationType:
          if (!settings.animation) changeBlock();
          else handleBodyOrHeadAnimationBlock();
          break;

        case speechType:
        case reflectionType:
        case readQuestionBlockType:
        case reflectionFormulaType:
          if (!settings.voice) changeBlock();
          else handleAudioBlock();
          break;

        case pauseType:
          handlePauseBlock();
          break;

        case feedbackBlockType:
          handleFeedbackBlock();
          break;

        default:
          break;
      }
  }, [state.currentData, state.currentBlockIndex]);

  useEffect(() => {
    changeIsAnimationOngoing(blocks.length !== 0);
  }, [questionId]);

  useEffect(() => {
    if (state.currentBlockIndex === blocks.length) {
      changeIsAnimationOngoing(false);
    }
  }, [state.currentBlockIndex, blocks.length]);

  const decideIfLoopAnimation = () =>
    (get(state, 'currentData.type', 'none') === speechType ||
      get(state, 'currentData.type', 'none') === reflectionType ||
      get(state, 'currentData.type', 'none') === readQuestionBlockType ||
      get(state, 'currentData.type', 'none') === reflectionFormulaType) &&
    get(state, 'currentData.isLoop', false);

  const getAnimationOptions = () => {
    if (
      settings.animation &&
      state.currentData &&
      state.currentData.animationData
    ) {
      const isSpeechType =
        state.currentData.type === speechType ||
        state.currentData.type === reflectionType ||
        state.currentData.type === readQuestionBlockType ||
        state.currentData.type === reflectionFormulaType;

      return {
        name: state.currentData.name,
        animationData: isSpeechType
          ? state.currentData.animationData[state.currentData.currentAnimation]
          : state.currentData.animationData,
      };
    }
    return {};
  };

  const defaultOptions = {
    renderer: 'svg',
    autoloadSegments: false,
    loop: decideIfLoopAnimation(),
    autoplay: false,
    ...getAnimationOptions(),
  };

  const isStopped =
    !state.currentData ||
    !state.currentData.animationData ||
    !decideIfPlaySpeechAnimation();

  return (
    <NarratorContainer>
      {settings.animation && (
        <Draggable disabled position={animationPos}>
          <Lottie
            ref={animationRef}
            options={defaultOptions}
            height={100}
            width={100}
            style={lottieStyles}
            isClickToPauseDisabled
            isStopped={isStopped}
          />
        </Draggable>
      )}
    </NarratorContainer>
  );
};

CharacterAnim.propTypes = {
  blocks: PropTypes.arrayOf(
    PropTypes.oneOfType([
      PropTypes.shape({ type: PropTypes.string, animation: PropTypes.string }),
      PropTypes.shape({
        type: PropTypes.string,
        text: PropTypes.string,
        audio_url: PropTypes.string,
      }),
    ]),
  ),
  questionId: PropTypes.string,
  settings: PropTypes.object,
  animationContainer: PropTypes.shape({
    clientWidth: PropTypes.number,
    clientHeight: PropTypes.number,
  }),
  previewMode: PropTypes.string,
  answers: PropTypes.object,
  changeIsAnimationOngoing: PropTypes.func,
  setFeedbackSettings: PropTypes.func,
  feedbackScreenSettings: PropTypes.object,
  audioInstance: PropTypes.object,
};

export default CharacterAnim;
