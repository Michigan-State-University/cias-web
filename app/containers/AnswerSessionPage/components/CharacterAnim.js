import React, { useEffect, useReducer } from 'react';
import PropTypes from 'prop-types';
import isNumber from 'lodash/isNumber';
import get from 'lodash/get';
import Lottie from 'react-lottie';
import { useDispatch } from 'react-redux';
import Draggable from 'react-draggable';
import { useIntl } from 'react-intl';

import useFeedbackHelper from 'utils/animationsHelpers/useFeedbackHelper';
import usePauseHelper from 'utils/animationsHelpers/usePauseHelper';
import useDidUpdateEffect from 'utils/useDidUpdateEffect';
import useMoveHelper from 'utils/animationsHelpers/useMoveHelper';
import useAudioHelper from 'utils/animationsHelpers/useAudioHelper';
import useAnimationHelper from 'utils/animationsHelpers/useAnimationHelper';
import { useAsync } from 'utils/useAsync';
import {
  bodyAnimationType,
  speechType,
  headAnimationType,
  reflectionType,
  readQuestionBlockType,
  pauseType,
  feedbackBlockType,
  reflectionFormulaType,
} from 'models/Narrator/BlockTypes';

import { setCurrentBlockIndex } from 'containers/AnswerSessionPage/actions';

import { NarratorContainer } from './styled';
import messages from '../messages';

const UPDATE = 'UPDATE';

const reducer = (state, action) => {
  const { type, newState } = action;
  switch (type) {
    case UPDATE:
      return { ...state, ...newState };
    default:
      return state;
  }
};

const initialState = {
  currentData: null,
  currentBlockIndex: 0,
};

const lottieStyles = {
  margin: 'none',
};

const CharacterAnim = ({
  blocks,
  questionId,
  settings,
  animationContainer,
  previewMode,
  answers,
  changeIsAnimationOngoing,
  setFeedbackSettings,
  feedbackScreenSettings: { sliderRef },
  audioInstance,
}) => {
  const { formatMessage } = useIntl();
  const [state, dispatch] = useReducer(reducer, initialState);
  const dispatchUpdate = (newState) =>
    dispatch({
      type: UPDATE,
      newState,
    });

  const globalDispatch = useDispatch();

  // actions
  const setCurrentBlockIndexAction = (index) =>
    globalDispatch(setCurrentBlockIndex(index));

  const changeBlock = async (prevIndex) => {
    clearAnimationBlock();

    cleanAudio();
    const nextIndex =
      (isNumber(prevIndex) ? prevIndex : state.currentBlockIndex) + 1;
    const nextBlock = blocks[nextIndex];
    if (nextBlock) {
      const newPosition = calculatePosition(
        nextBlock.action,
        nextBlock.endPosition,
      );

      await moveAnimation({ ...nextBlock, endPosition: newPosition });
      switch (nextBlock.type) {
        case bodyAnimationType:
        case headAnimationType:
          changeAnimation(nextBlock, nextIndex);
          break;

        case speechType:
        case reflectionType:
        case readQuestionBlockType:
        case reflectionFormulaType:
          changeSpeech(nextBlock, nextIndex);
          break;

        case pauseType:
          changePauseBlock(nextBlock, nextIndex);
          break;

        case feedbackBlockType:
          changeFeedback(nextBlock, nextIndex);
          break;

        default:
          await changeBlock(nextIndex);
          break;
      }
    } else {
      const nextAnim = getIdleAnimation();
      dispatchUpdate({
        currentData: nextAnim,
        currentBlockIndex: nextIndex,
      });
    }
  };

  const {
    getInitialBodyOrHeadAnimation,
    changeAnimation,
    handleBodyOrHeadAnimationBlock,
    getIdleAnimation,
    clearAnimationBlock,
    animationRef,
    fetchBodyAndHeadAnimations,
    loadedAnimations,
  } = useAnimationHelper(
    blocks,
    dispatchUpdate,
    changeBlock,
    state.currentData,
  );

  const { handlePauseBlock, getInitialPauseAnimation, changePauseBlock } =
    usePauseHelper(
      blocks,
      state.currentData,
      dispatchUpdate,
      changeBlock,
      getIdleAnimation,
    );

  const {
    changeSpeech,
    getInitialSpeechAnimation,
    cleanAudio,
    handleAudioBlock,
    decideIfPlaySpeechAnimation,
    fetchAudioAnimations,
    stopSpeech,
  } = useAudioHelper(
    blocks,
    dispatchUpdate,
    state.currentData,
    state.currentBlockIndex,
    animationRef.current,
    changeBlock,
    answers,
    settings,
    audioInstance,
  );

  const {
    getInitialFeedbackData,
    changeFeedback,
    handleFeedbackBlock,
    calculatePosition,
  } = useFeedbackHelper(
    blocks,
    dispatchUpdate,
    changeBlock,
    state.currentData,
    setFeedbackSettings,
    sliderRef,
    animationRef.current,
    loadedAnimations,
  );

  const { animationPos, moveAnimation, fetchMoveAnimations } = useMoveHelper(
    animationContainer,
    blocks,
    dispatchUpdate,
  );

  const getInitialData = () => {
    if (blocks.length) {
      switch (blocks[0].type) {
        case speechType:
        case reflectionType:
        case readQuestionBlockType:
        case reflectionFormulaType:
          return getInitialSpeechAnimation();

        case headAnimationType:
        case bodyAnimationType:
          return getInitialBodyOrHeadAnimation();

        case pauseType:
          return getInitialPauseAnimation();

        case feedbackBlockType:
          return getInitialFeedbackData();

        default:
          break;
      }
    }
  };
  const fetchData = async () => {
    if (blocks[0]) {
      const newPosition = calculatePosition(
        blocks[0].action,
        blocks[0].endPosition,
      );

      await moveAnimation({ ...blocks[0], endPosition: newPosition });
    }
    await fetchBodyAndHeadAnimations();
    await fetchAudioAnimations();
    await fetchMoveAnimations();
  };

  useAsync(
    fetchData,
    () => {
      setCurrentBlockIndexAction(-1);
      dispatchUpdate({
        currentData: getInitialData(),
        currentBlockIndex: 0,
      });
    },
    { deps: [questionId, previewMode], cleanUpFunction: stopSpeech },
  );

  useDidUpdateEffect(() => {
    if (state.currentData) {
      setCurrentBlockIndexAction(state.currentBlockIndex);

      switch (state.currentData.type) {
        case headAnimationType:
        case bodyAnimationType:
          if (!settings.animation) changeBlock();
          else handleBodyOrHeadAnimationBlock();
          break;

        case readQuestionBlockType:
        case speechType:
        case reflectionType:
        case reflectionFormulaType:
          if (!settings.voice) changeBlock();
          else handleAudioBlock();
          break;

        case pauseType:
          handlePauseBlock();
          break;

        case feedbackBlockType:
          handleFeedbackBlock();
          break;

        default:
          break;
      }
    }
  }, [state.currentData, state.currentBlockIndex]);

  useEffect(() => {
    changeIsAnimationOngoing(blocks.length !== 0);
  }, [questionId]);

  useEffect(() => {
    if (state.currentBlockIndex === blocks.length) {
      changeIsAnimationOngoing(false);
    }
  }, [state.currentBlockIndex, blocks.length]);

  const decideIfLoopAnimation = () =>
    (get(state, 'currentData.type', 'none') === speechType ||
      get(state, 'currentData.type', 'none') === reflectionType ||
      get(state, 'currentData.type', 'none') === readQuestionBlockType ||
      get(state, 'currentData.type', 'none') === reflectionFormulaType) &&
    get(state, 'currentData.isLoop', false);

  const getAnimationOptions = () => {
    if (
      settings.animation &&
      state.currentData &&
      state.currentData.animationData
    ) {
      const isSpeechType =
        state.currentData.type === speechType ||
        state.currentData.type === reflectionType ||
        state.currentData.type === readQuestionBlockType ||
        state.currentData.type === reflectionFormulaType;

      return {
        name: state.currentData.name,
        animationData: isSpeechType
          ? state.currentData.animationData[state.currentData.currentAnimation]
          : state.currentData.animationData,
      };
    }
    return {};
  };

  const defaultOptions = {
    renderer: 'svg',
    autoloadSegments: false,
    loop: decideIfLoopAnimation(),
    autoplay: false,
    ...getAnimationOptions(),
  };

  const isStopped =
    !state.currentData ||
    !state.currentData.animationData ||
    !decideIfPlaySpeechAnimation();

  return (
    <NarratorContainer>
      {settings.animation && (
        <Draggable disabled position={animationPos}>
          <Lottie
            ref={animationRef}
            options={defaultOptions}
            height={100}
            width={100}
            style={lottieStyles}
            isClickToPauseDisabled
            isStopped={isStopped}
            ariaLabel={formatMessage(messages.narratorAlt)}
          />
        </Draggable>
      )}
    </NarratorContainer>
  );
};

CharacterAnim.propTypes = {
  blocks: PropTypes.arrayOf(
    PropTypes.shape({
      action: PropTypes.string,
      endPosition: PropTypes.object,
      type: PropTypes.string,
    }),
  ),
  questionId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  settings: PropTypes.object,
  animationContainer: PropTypes.shape({
    clientWidth: PropTypes.number,
    clientHeight: PropTypes.number,
  }),
  previewMode: PropTypes.string,
  answers: PropTypes.object,
  changeIsAnimationOngoing: PropTypes.func,
  setFeedbackSettings: PropTypes.func,
  feedbackScreenSettings: PropTypes.object,
  audioInstance: PropTypes.object,
};

export default CharacterAnim;
