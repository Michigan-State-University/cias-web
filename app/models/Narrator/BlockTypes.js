import { colors } from 'theme';

export const bodyAnimationType = 'BodyAnimation';
export const headAnimationType = 'HeadAnimation';
export const speechType = 'Speech';
export const readQuestionBlockType = 'ReadQuestion';
export const reflectionType = 'Reflection';
export const reflectionFormulaType = 'ReflectionFormula';
export const pauseType = 'Pause';
export const feedbackBlockType = 'Feedback';

export const blockTypes = [
  speechType,
  bodyAnimationType,
  headAnimationType,
  readQuestionBlockType,
  pauseType,
  feedbackBlockType,
];

export const blockTypeToColorMap = {
  [bodyAnimationType]: colors.pink,
  [speechType]: colors.azure,
  [reflectionType]: colors.azure,
  [reflectionFormulaType]: colors.azure,
  [headAnimationType]: colors.golden,
  [readQuestionBlockType]: colors.electricViolet,
  [pauseType]: colors.coral,
  [feedbackBlockType]: colors.black,
};
