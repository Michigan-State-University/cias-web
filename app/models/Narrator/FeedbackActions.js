const showSpectrum = 'SHOW_SPECTRUM';
const showUserValue = 'SHOW_USER_VALUE';
const showLowerValue = 'SHOW_LOWER_VALUE';
const showHigherValue = 'SHOW_HIGHER_VALUE';
const noAction = 'NO_ACTION';

export const feedbackActions = {
  noAction,
  showSpectrum,
  showUserValue,
  showLowerValue,
  showHigherValue,
};
