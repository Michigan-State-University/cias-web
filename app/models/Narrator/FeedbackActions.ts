export enum EFeedbackAction {
  SHOW_SPECTRUM = 'SHOW_SPECTRUM',
  SHOW_USER_VALUE = 'SHOW_USER_VALUE',
  SHOW_LOWER_VALUE = 'SHOW_LOWER_VALUE',
  SHOW_HIGHER_VALUE = 'SHOW_HIGHER_VALUE',
  NO_ACTION = 'NO_ACTION',
}
