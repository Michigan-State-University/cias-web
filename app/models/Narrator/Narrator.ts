import { CamelToSnake } from 'global/types/camelToSnake';

import { NarratorBlockTypes } from './NarratorBlockTypes';

export interface Position {
  x: number;
  y: number;
}

export interface NarratorBlock {
  text?: Nullable<string[]>;
  type: NarratorBlockTypes;
  action?: string;
  sha256?: string[];
  animation: string;
  audioUrls?: string[];
  endPosition: Position;
  pauseDuration?: number;
  originalText?: Nullable<string[]>;
}

export interface NarratorSettings {
  voice: boolean;
  animation: boolean;
}

export interface Narrator {
  blocks: NarratorBlock[];
  settings: NarratorSettings;
}

export type NarratorDTO = CamelToSnake<Narrator>;
