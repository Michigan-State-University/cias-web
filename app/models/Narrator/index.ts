export * from './Narrator';
export * from './NarratorBlockTypes';
export * from './Animation';
export * from './FeedbackActions';
export * from './NarratorSettings';
export * from './utils';
