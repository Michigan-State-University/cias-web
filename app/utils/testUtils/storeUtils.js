import { createStore } from 'redux';
import createReducer from 'reducers';
import { initialState as authReducerInitialState } from 'global/reducers/auth/reducer';

export const createTestStore = (initialState) => {
  const reducer = (state) => state;

  const store = createStore(reducer, {
    auth: authReducerInitialState,
    ...initialState,
  });

  store.runSaga = () => {};
  store.injectedReducers = {};
  store.injectedSagas = {};
  store.createReducer = createReducer;

  return store;
};
