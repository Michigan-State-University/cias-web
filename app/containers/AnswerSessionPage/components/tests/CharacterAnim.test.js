/**
 *
 * Tests for CharacterAnim
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'i18n';

import { feedbackActions } from 'models/Narrator/FeedbackActions';
import { readQuestionBlockType } from 'models/Narrator/BlockTypes';
import { createTestStore } from 'utils/testUtils/storeUtils';

import CharacterAnim from '../CharacterAnim';

describe('<CharacterAnim />', () => {
  const store = createTestStore({});

  const mockedFunctions = {
    changeIsAnimationOngoing: jest.fn(),
    setFeedbackSettings: jest.fn(),
  };

  const defaultProps = {
    blocks: [
      {
        action: feedbackActions.noAction,
        animation: 'rest',
        audio_urls: ['/example/1'],
        endPosition: {
          x: 600,
          y: 600,
        },
        text: ['Enter main text/question for screen here'],
        type: readQuestionBlockType,
      },
    ],
    questionId: 'test',
    settings: {},
    animationContainer: { clientWidth: 0, clientHeight: 0 },
    previewMode: '',
    answers: {},
    feedbackScreenSettings: {},
    audioInstance: { clean: jest.fn(), stop: jest.fn() },
    ...mockedFunctions,
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <CharacterAnim {...defaultProps} />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <CharacterAnim {...defaultProps} />
        </IntlProvider>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
