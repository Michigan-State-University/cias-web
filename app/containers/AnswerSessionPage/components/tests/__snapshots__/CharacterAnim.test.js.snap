// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterAnim /> Should render and match the snapshot 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  height: 0;
  width: 0;
}

.c0 > div {
  -webkit-transition: all 1480ms cubic-bezier(0.65,0.05,0.36,1);
  transition: all 1480ms cubic-bezier(0.65,0.05,0.36,1);
}

<div>
  <div
    class="c0"
  />
</div>
`;
